kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -143,6 +145,13 @@ class BatchJobSubmission(
   private def setStateIfNotCanceled(newState: OperationState): Unit = state.synchronized {
     if (state != CANCELED) {
       setState(newState)
+      applicationInfo.filter(_.id != null).foreach { ai =>
+        session.getSessionEvent.foreach(_.engineId = ai.id)
+      }
+      if (newState == RUNNING) {
+        session.onEngineOpened()
```

Comment:
Will this be called multiple times?

Reply:
One time is expected

Reply:
check the session opened time to confirm that